openslides_backend/action/actions/user/save_saml_account.py

```diff
@@ -157,7 +184,45 @@ def set_defaults(self, instance: dict[str, Any]) -> dict[str, Any]:
         if "is_physical_person" not in instance:
             instance["is_physical_person"] = True
         instance["can_change_own_password"] = False
-        instance["organization_id"] = ONE_ORGANIZATION_ID
         instance["username"] = self.generate_usernames([instance.get("saml_id", "")])[0]
-        instance["meta_new"] = True
-        return super().set_defaults(instance)
+        return instance
+
+    def check_for_group_add(self) -> tuple[int, int] | tuple[None, None]:
+        NoneResult = (None, None)
+        if not (
+            meeting_info := cast(dict, self.saml_attr_mapping.get("meeting"))
+        ) or not (external_id := meeting_info.get("external_id")):
+            return NoneResult
+
+        meetings = self.datastore.filter(
+            collection="meeting",
+            filter=FilterOperator("external_id", "=", external_id),
+            mapped_fields=["id", "default_group_id"],
+        )
+        if len(meetings) == 1:
+            meeting = next(iter(meetings.values()))
+            group_id = meeting["default_group_id"]
+        else:
+            self.logger.warning(
+                f"save_saml_account found {len(meetings)} meetings with external_id {external_id}"
+            )
+            return NoneResult
+        if external_group_id := meeting_info.get("external_group_id"):
+            groups = self.datastore.filter(
+                collection="group",
+                filter=And(
+                    [
+                        FilterOperator("external_id", "=", external_group_id),
+                        FilterOperator("meeting_id", "=", meeting.get("id")),
+                    ]
+                ),
+                mapped_fields=["id"],
+            )
+            if len(groups) == 1:
```

Comment:
Maybe log a warning in the other case? (that the default group is used because the external group id was not found)